### ISSUE TITLE
Search holds on to last search result

### ISSUE THREAD

**Issue by little-wow** [creativecommons/cccatalog-frontend#358]
see GIF

![cc-search](https://user-images.githubusercontent.com/7093838/56959954-43f71280-6b1d-11e9-986c-ab48785d8ca0.gif)

**Comment by janeatcc** [comment]
can you say a little more about this @little-wow ? all I see in the gif is you searching for "moon phases"  ?

**Comment by little-wow** [comment]
I searched for "flowers" before I searched for "moon phases" and there's a brief flash of flower results before the moon phases

**Comment by janeatcc** [comment]
hm, that may just be a remnant, but not a bug. adding @aldenstpage to investigate

**Comment by aldenstpage** [comment]
This looks like a front end issue to me, likely that the app is retaining old searches until they are overwritten with new results. There's no relationship between individual search queries on the backend.

**Comment by brenoferreira** [comment]
It's a frontend issue. I'll look into the cause. Thanks for spotting this @little-wow

**Comment by elharony** [comment]
I agree with you, @aldenstpage . It seems like a Front end issue!
Can I work on it?

**Comment by elharony** [comment]
Oh... I didn't see your comment @brenoferreira (almost 5 seconds late)
If it's not an urgent bug; Can you give me a one-day to fix it? If I can't fix it, I will tell you what I've found (i.e. Brief Report)!

**Comment by brenoferreira** [comment]
Sounds good

**Comment by elharony** [comment]
Hi @brenoferreira ,
Unfortunately, I couldn't fix it by myself but I've listed all my thoughts & walkthrough to shorten the estimated work time for you to fix it.

---

In the `SearchGrid.vue`, we reset everything if the Search is changed:
```js
// Line: 156
searchChanged() {
  this.showGrid = false;
  this.$store.commit(SET_IMAGES, { images: [] });
  this.currentPage = 0;
  this.$nextTick(() => {
    this.$refs.infiniteLoader.$emit('$InfiniteLoading:reset');
  });
},
```
But, we don't apply this function if the user get redirected to another page. Well, I am not sure if we should even do it!

Here is a user story to explain it better:

## User Story
1. Start from the `homepage` search
2. Search for a specific `search term` (i.e. Flowers), and get redirected to the `search results` page 
3. Go back to the `homepage` search
4. Search for another `search term` (i.e. moon phases), and get redirected to the `search results` page
5. Now, the `search results` page saved a copy of our (2) search.

### Why we need to clear the `SearchGrid.vue` if redirected to another page?
To fix the previously mentioned issue.

### But, why we need to NOT clear the `SearchGrid.vue` if redirected to another page?
Because the user might get redirected to a page like `PhotoDetailPage.vue` as the user might **go back** to see the results and click on a different image... If we cleared the `SearchGrid.vue`, the user will get an empty grid!

### Is there a possible solution?
Yes. I think if we could `reset` the `SearchGrid.vue` if the user gets redirected to the `HomePage.vue` **ONLY**, to avoid the mentioned issue above if s/he gets redirected to `PhotoDetailPage.vue` or any other page, that would fix this problem.

---

Those are my thoughts/report about this issue, please let me know if I could help!

**Comment by brenoferreira** [comment]
Hi @elharony 

Thank you for your bug investigation. I took your insights and developed a fix you can see at #382

**Comment by elharony** [comment]
You did a great job here, @brenoferreira ! 👋
